qutip/tests/solver/heom/test_bofin_baths.py

```diff
@@ -463,3 +472,247 @@ def test_create(self):
+        a2, b2, c2 = _leastsq(
+            self.spectral_density,
+            C,
+            t,
+            guesses=guesses,
+            lower=lower,
+            upper=upper,
+            sigma=sigma,
+        )
+        C2 = self.spectral_density(t, a2, b2, c2)
+        assert np.isclose(C, C2).all()
+
+    def test_fit(self):
+        a, b, c = [list(range(2))] * 3
+        w = np.linspace(0.1, 10 * 5, 1000)
+        J = self.spectral_density(w, a, b, c)
+        rmse, [a2, b2, c2] = _fit(self.spectral_density, J, w, N=2)
+        J2 = self.spectral_density(w, a2, b2, c2)
+        assert np.isclose(J, J2).all()
+        assert rmse < 1e-15
+
+
+class TestFitSpectral:
+    alpha = 0.005
+    wc = 1
+    T = 1
+    bath = FitSpectral(T, sigmax(), Nk=2)
+
+    def test_spectral_density_approx(self):
+        J = 0.4
+        a, b, c = [list(range(2))] * 3
+        w = 1
+        assert self.bath._spectral_density_approx(w, a, b, c) == J
+        a, b, c = [list(range(3))] * 3
+        w = 2
+        assert self.bath._spectral_density_approx(w, a, b, c) == J
+
+    def test_spec_spectrum_approx(self):
+        a, b, c = [list(range(2))] * 3
+        w = np.linspace(0.1, 10 * self.wc, 1000)
+        J = self.bath._spectral_density_approx(w, a, b, c)
+        pow = J * ((1 / (np.e ** (w / self.T) - 1)) + 1) * 2
+        rmse, self.bath.params_spec = _fit(
+            self.bath._spectral_density_approx, J, w, N=2, label="spectral"
+        )
+        pow2 = self.bath.power_spectrum(w, 1/self.bath.T)
+        assert np.isclose(pow, pow2).all()
+
+    def test_get_fit(self):
+        a, b, c = [list(range(5))] * 3
+        w = np.linspace(0.1, 10 * self.wc, 1000)
+        J = self.bath._spectral_density_approx(w, a, b, c)
+        self.bath.get_fit(J, w)
+        a2, b2, c2 = self.bath.params_spec
+        J2 = self.bath._spectral_density_approx(w, a2, b2, c2)
+        assert np.sum(J - J2) < (1e-12)
+        assert self.bath.spec_n <= 5
+
+    def test_matsubara_coefficients_from_spectral_fit(self):
+        ckAR, vkAr, ckAI, vkAI = (
+            [
+                (0.29298628613108785 - 0.2097848947562078j),
+                (0.29298628613108785 + 0.2097848947562078j),
+                (-0.01608448645347298 + 0j),
+                (-0.002015397620259254 + 0j),
+            ],
+            [(1 - 1j), (1 + 1j), (6.283185307179586 + 0j),
+             (12.566370614359172 + 0j)],
+            [(-0 + 0.25j), -0.25j],
+            [(1 - 1j), (1 + 1j)],
+        )
+        self.bath.params_spec = np.array([1]), np.array([1]), np.array([1])
+        self.bath._matsubara_spectral_fit()
+        ckAR2, vkAr2, ckAI2, vkAI2 = self.bath.cvars
+        assert np.isclose(ckAR, ckAR2).all()
+        assert np.isclose(vkAr, vkAr2).all()
+        assert np.isclose(ckAI, ckAI2).all()
+        assert np.isclose(vkAI, vkAI2).all()
+
+
+class TestFitCorr:
+    alpha = 0.005
+    wc = 1
+    T = 1
+    bath = FitCorr(sigmax())
```

Comment:
Just move these into the tests that need them since they're only one extra line usually. Alternatively, write a pytest fixture if you need common setup for tests.

Reply:
Sure, will do!